### ISSUE TITLE
Authentication with credential acquired in cloud shell fails

### ISSUE THREAD

**Issue by wpbrown** [Azure/azure-cli#14218]
**Describe the bug**
Trying to use a credential retrieved from the CLI fails when using Cloud Shell MSI because the token format  does not match the code.

**To Reproduce**
```python
from azure.common.credentials import get_azure_cli_credentials
credential, _ = get_azure_cli_credentials(resource="https://storage.azure.com/")
```

Use the credential in the blob client...
```
  File "/.../site-packages/azure/core/pipeline/policies/_authentication.py", line 93, in on_request
    self._token = self._credential.get_token(*self._scopes)
  File "/.../site-packages/azure/cli/core/adal_authentication.py", line 64, in get_token
    return AccessToken(token, int(full_token['expiresIn'] + time.time()))
KeyError: 'expiresIn'
```

**Expected behavior**
`adal_authentication.get_token` returns a valid `AccessToken`.

**Environment summary**
`azure_cli_core-2.8.0` in Azure Cloud Shell.

**Additional context**

The actual full_token:
```json
{
    "access_token": "{REMOVED}",
    "refresh_token": "",
    "expires_in": "3900",
    "expires_on": "1593791553",
    "not_before": "1593787653",
    "resource": "https://storage.azure.com/",
    "token_type": "Bearer"
}
```
The code looks for `expiresIn`, but under cloud shell the full_token contains `expires_in` and `expires_on`.

**Comment by yonzhan** [comment]
add to S173

**Comment by jiasli** [comment]
This should have been fixed in #14187.
